instrumentation-core/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/httpclients/resttemplate/RestTemplateAutoConfiguration.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.httpclients.resttemplate;
+
+import io.opentelemetry.instrumentation.spring.autoconfigure.httpclients.HttpClientProperties;
+import io.opentelemetry.trace.Tracer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.client.RestTemplate;
+
+/** Configures RestTemplateBeanPostProcessor bean */
+@Configuration
+@ConditionalOnClass(RestTemplate.class)
+@EnableConfigurationProperties(HttpClientProperties.class)
+@ConditionalOnProperty(prefix = "opentelemetry.trace.httpclients", name = "enabled")
+public class RestTemplateAutoConfiguration {
+
+  @Autowired private Tracer tracer;
+
+  @Bean
+  public RestTemplateBeanPostProcessor restTemplateBeanPostProcessor() {
```

Comment:
consider a more unique name for the method name (which translates to the name of the bean itself). Or, use additional annotation to rename the bean name.

Reply:
I added the prefix `otel` to all bean names